app/routes/_layout+/assets.$assetId.overview.tsx

```diff
@@ -129,6 +133,33 @@ export async function loader({ context, request, params }: LoaderFunctionArgs) {
       asset.bookings = [currentBooking];
     }
 
+    /** We only need customField with same category of asset or without any category */
+    let customFields = asset.categoryId
+      ? asset.customFields.filter(
+          (cf) =>
+            !cf.customField.categories.length ||
+            cf.customField.categories
+              .map((c) => c.id)
+              .includes(asset.categoryId!)
+        )
+      : asset.customFields;
+
+    customFields = customFields.map((cf) => {
```

Comment:
Shouldnt all of this go inside `getCustomFieldDisplayValue`, as that is what its supposed to do, extract the value in special cases. For now it was used only for date but this is another good use-case.